src/ContentClient.ts

```diff
@@ -186,10 +186,11 @@ export class ContentClient implements ContentAPI {
     return retry(
       async () => {
         const content = await this.fetcher.fetchBuffer(`${this.contentUrl}/contents/${contentHash}`, timeout)
-        const downloadedHash = await Hashing.calculateBufferHash(content)
+        const downloadedHashAsV3 = await Hashing.calculateBufferHash(content)
+        const downloadedHashAsV4 = await Hashing.calculateIPFSHash(content)
```

Comment:
How often is this function used? could we benefit from something like this?

```ts
const hash = contentHash.startsWith('Qm')
  ? await Hashing.calculateBufferHash(content)
  : await Hashing.calculateBufferHash(content)
```

Reply:
Yes!